packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -690,21 +697,30 @@ class _CupertinoBackGestureDetectorState<T> extends State<_CupertinoBackGestureD
                            MediaQuery.paddingOf(context).left :
                            MediaQuery.paddingOf(context).right;
     dragAreaWidth = max(dragAreaWidth, _kBackGestureWidth);
-    return Stack(
-      fit: StackFit.passthrough,
-      children: <Widget>[
-        widget.child,
-        PositionedDirectional(
-          start: 0.0,
-          width: dragAreaWidth,
-          top: 0.0,
-          bottom: 0.0,
-          child: Listener(
-            onPointerDown: _handlePointerDown,
-            behavior: HitTestBehavior.translucent,
+    return PopScope(
```

Comment:
@goderbauer Is there a good way for a route to be informed if it's going to be popped or pushed over? This PopScope setup works for pops, but ideally I want to know if the route has something pushed on top of it, too. I tried watching for changes in PageRoute.isCurrent, but it comes in too late, after the recognizer thinks the drag has ended.

Reply:
I don't fully understand what you need or are trying to do... Routes have `didChangeNext` that informs them if something is added (e.g. pushed) on top of them, but I don't know if this comes early enough?

Reply:
Thanks for the suggestion. I tried didChangeNext/Previous by adding a setState in them when I needed to rebuild, but it didn't rebuild fast enough. Actively reading isCurrent/isActive on the route instead of waiting for it to be passed down seems to be fast enough, though 👍 .